### PR TITLE
Bishop moves

### PR DESCRIPTION
**Pull request by brephelps**
# Trello Ticket: 
https://trello.com/c/bozUWKAl/26-validmove-for-the-bishop

# Description:
Valid moves for Bishop. 

# Documentation Referenced:

### CODE REVIEW

**Comment by brephelps** [comment]
I actually need to add some code to check for the qualifying locations within the board, so ignore for now.

File: spec/models/bishop_spec.rb [hunk 1]
```diff
@@ -1,4 +1,21 @@
 require 'rails_helper'
 
 RSpec.describe Bishop, type: :model do
```
Comment (ivanperez002) [inline spec/models/bishop_spec.rb]: Looking very good. So far I think we can get this merged into the master.

Comment (ivanperez002) [reply spec/models/bishop_spec.rb]: Only suggestion, I would to be able to test valid move with the obstructed method. Maybe  create a separate RSPEC text, that makes a game, populates the game, then it moves the Pawn that is diagonal from a Bishop to be moved? Maybe move the pawn up one, and thus the test should still test true as working as the bishop is not blocked in that direction? But doing the other diagonal direction the test would false, as the pawn is there?

Comment (brephelps) [reply spec/models/bishop_spec.rb]: I don't mind rewriting the code that way, but I'm also thinking that the is_obstructed? tests should be separate and testing specifically that method. @R-Currie @ivanperez002 @brihong - thoughts?

Comment (brihong) [reply spec/models/bishop_spec.rb]: Great suggestions to both @brephelps @ivanperez002 ! I would agree with Bre's approach to test the is_obstructed? separately in chess_piece.spec.rb 
I've shared my struggles with trying to pass the obstructions test in the rook_spec.rb heh 
However, Ivan if you were able to write a test for the Queen's obstruction, please share

**Pull request merged**